src/postgres/commands/checkAuthentication.ts

```diff
@@ -24,10 +29,22 @@ export async function checkAuthentication(context: IActionContext, treeItem: Pos
                 // Need to configure firewall only for Azure Subscritption accounts
             } else if (treeItem.parent.resourceGroup && parsedError.errorType === firewallNotConfiguredErrorType) {
                 await configurePostgresFirewall(context, treeItem.parent);
+            } else if (treeItem.parent.resourceGroup && parsedError.errorType === 'ETIMEDOUT' && !(await isFirewallRuleSet(treeItem.parent))) {
+                await configurePostgresFirewall(context, treeItem.parent);
             } else {
                 throw error;
             }
         }
     }
     return clientConfig;
 }
+
+export async function isFirewallRuleSet(treeItem: PostgresServerTreeItem): Promise<boolean> {
+    const serverType: PostgresServerType = nonNullProp(treeItem, 'serverType');
+    const client = createAbstractPostgresClient(serverType, treeItem.root);
+    client.firewallRules.listByServer
+    const result: FirewallRuleListResult = (await client.firewallRules.listByServer(nonNullProp(treeItem, 'resourceGroup'), nonNullProp(treeItem, 'azureName')))._response.parsedBody;
+    const publicIp: string = await getPublicIp();
+    return (result.some(async value => value.startIpAddress <= publicIp && publicIp <= value.endIpAddress));
```

Comment:
The ip addresses are not numbers, so I don't think we can compare like this. Just checking equality is probably good enough. The only other option is to parse and compare each subsection, but I'm not sure that extra logic is worth it.

Reply:
That's true!